hackathon_site/dashboard/frontend/src/slices/users/userSlice.test.ts

```diff
@@ -110,13 +156,13 @@ describe("fetchUserData Thunk and Reducer", () => {
     });
 
     describe("Failed to fetch user data", () => {
-        beforeAll(() => {
+        beforeEach(() => {
             // Used by displaySnackbar
             jest.spyOn(global.Math, "random").mockReturnValue(0.45526621894095487);
```

Comment:
What does line 161 do? I'm not sure how the test works

Reply:
so beforeEach(), afterEach() and similar functions are not actually tests, they are functions that run code before or after each unit test. Here, jest.spyOn() "spies on" the Math object and mocks the random function so that Math.random() will always return 0.45526621894095487 whenever used in the code.